src/impl/load.c

```
@@ -0,0 +1,289 @@
+            return NULL;
+        }
+
+        int match_result = regexec(&regex, identifier, 2, matches, 0);
+        if (match_result == 0) {
+            int start = matches[1].rm_so;
+            int end = matches[1].rm_eo;
+            int id_len = end - start;
+            
+            game_id_str = malloc_or_die(id_len + 1);
+            strncpy(game_id_str, identifier + start, id_len);
+            game_id_str[id_len] = '\0';
+        } else {
+            regfree(&regex);
+            error_stack_push(error_stack, ERROR_STATUS_CONFIG_LOAD_MISSING_ARG,
+                            get_formatted_string("Failed to extract game ID from woogles URL: %s", identifier));
+            return NULL;
+        }
+
+        regfree(&regex);
+    } else {
+        // Check if it's a standalone woogles game ID (must be alphanumeric only)
+        if (is_all_digits_or_empty(identifier)) {
+            // All digits likely means cross-tables, not woogles
+            return NULL;
+        }
+        // Check if identifier is alphanumeric only
+        for (int i = 0; identifier[i]; i++) {
+            if (!isalnum(identifier[i])) {
+                // Contains non-alphanumeric characters, not a woogles game ID
+                return NULL;
+            }
+        }
+        // It's alphanumeric and not all digits, assume it's a woogles game ID
+        game_id_str = string_duplicate(identifier);
+    }
+    
+    char *gcg_content = NULL;
+
+    if (game_id_str) {
+        // Use woogles API to get GCG content
+        char *curl_cmd = get_formatted_string("curl -s -H 'Content-Type: application/json' "
+                                              "https://woogles.io/api/game_service.GameMetadataService/GetGCG "
+                                              "-d '{\"game_id\":\"%s\"}'", game_id_str);
+        
+        char *response = get_process_output(curl_cmd);
+        free(curl_cmd);
+        free(game_id_str);
+        
+        if (response) {
```

If the response is NULL we should just return an error which avoids the nesting.